src/components/popover/popover.jsx

```diff
@@ -0,0 +1,94 @@
+import PropTypes from 'prop-types';
+import React, {
+  forwardRef,
+  useEffect,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from 'react';
+import IconButton from '../icon-button/icon-button.jsx';
+import iconClear from '../../svgs/clear.svg';
+import styles from './popover.css';
+
+const Popover = forwardRef(function Popover(props, ref) {
+  const [open, setOpen] = useState(false);
+  const closeIconRef = useRef();
+  const backupRef = useRef();
+  const selfRef = ref || backupRef;
+
+  const onWindowClick = () => {
+    if (open) {
+      setOpen(false);
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener('click', onWindowClick);
+
+    if (closeIconRef.current && open) {
+      closeIconRef.current.focus();
+    }
+
+    if (!open) {
+      props.onClose();
+    }
+
+    return () => {
+      window.removeEventListener('click', onWindowClick);
+    };
+  }, [open]);
+
+  useImperativeHandle(selfRef, () => ({
+    get open() {
+      return open;
+    },
+    set open(value) {
+      setOpen(value);
+    },
+  }));
+
+  if (!open) {
+    return null;
+  }
+
+  return (
+    // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
+    <section
+      aria-labelledby="popover-heading"
+      className={styles.container}
+      onClick={(event) => { event.stopPropagation(); }}
+      role="dialog"
+      style={{ left: `${props.left}px`, top: `${props.top}px` }}
```

Comment:
Just food for thought: 

Since we want a separation of CSS vs JS, we've implemented a "bridge" by setting CSS variables. 

```suggestion
      style={{ '--left': props.left, '--top': props.top }}
```

And the CSS file will be self-contained:

```css
.foo {
  left: var(--left);
  top: var(--top);
}
```

Reply:
I'll keep this in mind for next time, but with the move to `absolute`, this props API doesn't exist anymore. 😄